### PR TITLE
performance: avoid desktop/mobile menus unnecesary renderings

### PR DESCRIPTION
**Pull request by nhsz**
This PR avoids unnecesary nav menus renderings on desktop/mobile

## Description

- avoids rendering desktop Menu version on mobile
- avoids rendering desktop LanguagePicker version on mobile
- extracts `MobileCloseBar` component and renders on mobile only
- moves `LanguagePicker` handlers to external functions to avoid re-creating them on each render

### CODE REVIEW

**Review by pettinarip — COMMENTED** [review]
@nhsz good job, left a couple of comments.

File: src/components/LanguagePicker/index.tsx [hunk 1]
```diff
@@ -59,6 +63,39 @@ const LanguagePicker = ({
...
+      eventAction: "Locale chosen",
+      eventName: displayInfo.localeOption,
+    })
+  const handleBaseLinkClose = () =>
+    onClose({
+      eventAction: "Translation program link (menu footer)",
+      eventName: "/contributing/translation-program",
+    })
+
+  const MobileCloseBar = () => (
```
[the remark this answers is not in the document]
Comment (nhsz) [reply src/components/LanguagePicker/index.tsx:79]: Fixed

File: src/components/Nav/Menu/index.tsx [hunk 2]
```diff
@@ -78,32 +82,37 @@ const Menu = ({ sections, ...props }: NavMenuProps) => {
...
-                        items={items}
-                        activeSection={activeSection}
-                        onClose={onClose}
-                      />
-                    </Box>
-                  </NavigationMenu.Content>
+
+                  {/* avoid rendering desktop menu on mobile */}
+                  {/* Desktop Menu content */}
+                  {isDesktop && (
```
[the remark this answers is not in the document]
Comment (nhsz) [reply src/components/Nav/Menu/index.tsx]: Right, fixed, thanks

File: src/components/Nav/Menu/index.tsx [hunk 3]
```diff
@@ -78,6 +78,9 @@ const Menu = ({ sections, ...props }: NavMenuProps) => {
                       </Text>
                     </Button>
                   </NavigationMenu.Trigger>
+
+                  {/* avoid rendering desktop menu on mobile */}
+                  {/* Desktop Menu content */}
```
Comment (pettinarip) [inline src/components/Nav/Menu/index.tsx:83]: cleaning up comments
```suggestion
```

**Review by pettinarip — APPROVED** [review]

**Pull request merged**